pkg/radar/threat.go

```diff
@@ -0,0 +1,49 @@
+package radar
+
+import (
+	"github.com/dharmab/skyeye/pkg/bearings"
+	"github.com/dharmab/skyeye/pkg/brevity"
+	"github.com/dharmab/skyeye/pkg/coalitions"
+	"github.com/martinlindhe/unit"
+	"github.com/paulmach/orb/geo"
+)
+
+func (s *scope) Threats(coalition coalitions.Coalition) map[uint32][]brevity.Group {
+	threats := make(map[uint32][]brevity.Group)
+	itr := s.contacts.itr()
+	for itr.next() {
+		trackfile := itr.value()
+
+		if trackfile.Contact.Coalition == coalition {
+			continue
+		}
+
+		if !isValidTrack(trackfile) {
+			continue
+		}
+
+		groups := s.findNearbyGroups(trackfile.LastKnown().Point, 0, 100000, 100*unit.NauticalMile, coalition, brevity.FixedWing)
+
+		threatGroups := []brevity.Group{}
+		for _, group := range groups {
+			origin := trackfile.LastKnown().Point
+			bearing := bearings.NewTrueBearing(unit.Angle(geo.Bearing(origin, group.point())) * unit.Degree).Magnetic(s.Declination(origin))
+			_range := unit.Length(geo.Distance(origin, group.point()))
+			altitude := group.Altitude()
+			aspect := brevity.AspectFromAngle(bearing, group.course())
+			braa := brevity.NewBRAA(bearing, _range, altitude, aspect)
+			group.braa = braa
+			group.bullseye = nil
+
+			// TODO factor in aircraft threat range
+			if _range < brevity.MandatoryThreatDistance || aspect == brevity.Hot {
```

Comment:
I would suggest 25nm as a default for now, 35nm is quite far. For example, that's the shot range on my timeline.

Reply:
This probably needs to be configurable as it's usually based on mission briefing, but ACC recommends a 35NM default:

>  Controllers should use 35 nm, independent of aspect, as a default THREAT criteria.

'course this is something that's easy to be pragmatic about.

Reply:
That's also an IRL suggestion based on IRL weapons which are probably way more effective in DCS.

Reply:
Range is now configurable